src/Platform/Microsoft.Testing.Platform/Services/CTRLPlusCCancellationTokenSource.cs

```diff
@@ -11,14 +11,18 @@ internal sealed class CTRLPlusCCancellationTokenSource : ITestApplicationCancell
     private readonly CancellationTokenSource _cancellationTokenSource = new();
     private readonly ILogger? _logger;
 
-    public CTRLPlusCCancellationTokenSource(IConsole? console = null, ILogger? logger = null)
+    public CTRLPlusCCancellationTokenSource(IConsole? console = null, ILogger? logger = null, PoliciesService? policiesService = null)
     {
         if (console is not null)
         {
             console.CancelKeyPress += OnConsoleCancelKeyPressed;
         }
 
         _logger = logger;
+        if (policiesService is not null)
+        {
+            _cancellationTokenSource.Token.Register(() => policiesService.AbortPolicy.ExecuteCallbacksAsync(cancellationToken: default).GetAwaiter().GetResult());
```

Comment:
```suggestion
            _cancellationTokenSource.Token.Register(async () => await policiesService.AbortPolicy.ExecuteCallbacksAsync(cancellationToken: default));
```

Reply:
@Evangelink That will be fire-and-forget. CancellationToken.Register doesn't have a `Func<Task>` overload 😕

Reply:
effectively same as `async void`